File: Ruby/Cliente.rb
module Veterinaria
    class Cliente
        def initialize nombre
            @nombre = nombre
            @mascotas = Array.new
        end

        def agregarMascota mascota
            @mascotas += [mascota]
        end

        def reporteMascotas
            mensaje = ""
            @mascotas.each do |mascota|
                mensaje += mascota.presentarse() + "\n"
            end
            return mensaje
        end
    end
end

File: PHP/Cliente.php
<?php

require_once 'Mascota.php';

class Cliente {
    private $nombre;
    private $mascotas = array();

    public function __construct($nombre) {
        $this->nombre = $nombre;
    }

    public function agregarMascota($mascota) {
        $this->mascotas[] = $mascota;
    }

    public function reporteMascotas() {
        $mensaje = "";
        foreach ($this->mascotas as $mascota) {
            $mensaje = $mensaje . $mascota->presentarse() . "\n";
        }
        return $mensaje;
    }

}
